dom0/sd-dom0-files.sls

```diff
@@ -6,6 +6,68 @@
 # over time. These scripts should be ported to an RPM package.
 ##
 
+dom0-rpm-test-key:
+  file.managed:
+    # We write the pubkey to the repos config location, because the repos
+    # config location is automatically sent to dom0's UpdateVM. Otherwise,
+    # we must place the GPG key inside the fedora-29 TemplateVM, then
+    # restart sys-firewall.
+    - name: /etc/pki/rpm-gpg/RPM-GPG-KEY-securedrop-workstation-test
+    - source: "salt://sd/sd-workstation/apt-test-pubkey.asc"
+    - user: root
+    - group: root
+    - mode: 644
+
+dom0-rpm-test-key-import:
+  cmd.run:
+    - name: sudo rpm --import /etc/pki/rpm-gpg/RPM-GPG-KEY-securedrop-workstation-test
+    - require:
+      - file: dom0-rpm-test-key
+
+dom0-rpm-test-key-sys-firewall:
+  cmd.run:
+    # Pass in the pubkey directly to sys-firewall, so it's available on the
+    # UpdateVM for dom0 while we're configuring dom0 repos.
+    - name: >
+        qvm-run -p sys-firewall '
+        sudo tee /etc/pki/rpm-gpg/RPM-GPG-KEY-securedrop-workstation-test'
+        < /srv/salt/sd/sd-workstation/apt-test-pubkey.asc
+
+dom0-rpm-test-key-sys-firewall-import:
+  cmd.run:
+    - name: >
+        qvm-run --no-gui sys-firewall '
+        sudo rpm --import /etc/pki/rpm-gpg/RPM-GPG-KEY-securedrop-workstation-test'
+    - require:
+      - cmd: dom0-rpm-test-key-sys-firewall
+
+dom0-workstation-rpm-repo:
+  # We use file.managed rather than pkgrepo.managed, because Qubes dom0
+  # settings write new repos to /etc/yum.real.repos.d/, but only /etc/yum.repos.d/
+  # is copied to the UpdateVM for fetching dom0 packages.
+  file.managed:
+    - name: /etc/yum.repos.d/securedrop-workstation-dom0.repo
+    - user: root
+    - group: root
+    - mode: 644
+    - contents: |
+        [securedrop-workstation-dom0]
+        gpgcheck=1
+        gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-securedrop-workstation-test
+        enabled=1
+        baseurl=https://dev-bin.ops.securedrop.org/dom0-rpm-repo/
```

Comment:
Hey @conorsch - you probably want an over-ride variable here, we dont want to target `dev-bin.ops.securedrop.org` in all scenarios, do we?

Reply:
Right now, we _only_ need dev support, but you're right that we'll eventually want to flip this to dev—and preserve the ability for developers to override. Any thoughts on how to do that cleanly in Salt? We're not currently using any vars-based configuration, as we do heavily over in SecureDrop core (https://github.com/freedomofpress/securedrop/).

Reply:
Yea @conorsch -  [look into pillars here](https://docs.saltstack.com/en/latest/topics/tutorials/pillar.html). Variable interpolation uses the jinja syntax and you can inject complex jinja logic where-ever (one of the pros/cons in using salt v. ansible).